### PR TITLE
scatter_add and scatter_nd_add

### PR DESCRIPTION
**Pull request by silee2**

### CODE REVIEW

File: src/ngraph/op/scatter_add.cpp [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+    NODE_VALIDATION_CHECK(this,
+                          inputs_shape.rank().is_dynamic() ||
+                              indices_shape.rank().is_dynamic() ||
+                              updates_shape.rank().is_dynamic() ||
+                              static_cast<size_t>(updates_shape.rank()) ==
+                                  static_cast<size_t>(indices_shape.rank()) +
+                                  static_cast<size_t>(inputs_shape.rank()) - 1,
+                          "Updates rank is expected to be indices rank + inputs rank - 1");
+
+    // TODO: updates shape must be indices shape + inputs shape[1:]
```
Comment (aprocter) [inline src/ngraph/op/scatter_add.cpp]: Any chance we could go ahead and implement this check?

File: src/ngraph/op/scatter_nd_add.cpp [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+    NODE_VALIDATION_CHECK(
+        this,
+        inputs_shape.rank().is_dynamic() || indices_shape.rank().is_dynamic() ||
+            updates_shape.rank().is_dynamic() ||
+            static_cast<size_t>(updates_shape.rank()) ==
+                static_cast<size_t>(indices_shape.rank()) + static_cast<size_t>(inputs_shape.rank()) -
+                static_cast<size_t>(indices_shape[static_cast<size_t>(indices_shape.rank()) - 1]) - 1,
+        "Rank of updates must be rank of inputs + rank of indices - last dimension of indices - 1");
+
+    // TODO: check updates shape
```
Comment (aprocter) [inline src/ngraph/op/scatter_nd_add.cpp]: Any chance we could go ahead and implement this check?

File: src/ngraph/op/scatter_add.cpp (hunk 1 above)
Comment (diyessi) [reply src/ngraph/op/scatter_add.cpp]: Yes.

Comment (silee2) [reply src/ngraph/op/scatter_add.cpp]: Done

File: src/ngraph/op/scatter_nd_add.cpp (hunk 2 above)
Comment (silee2) [reply src/ngraph/op/scatter_nd_add.cpp]: Done

**Pull request merged**